### PR TITLE
Update external-secrets docs

### PR DESCRIPTION
**Pull request by moelsayed**
**Description**

Changes proposed in this pull request:

- Update external-secrets docs

**Related issue(s)**

### CODE REVIEW

**Review by Halamix2 — APPROVED** [review]

File: prow/cluster/resources/external-secrets/README.md [hunk 1]
```diff
@@ -70,6 +70,10 @@ spec:
       version: latest # version of the GCP Secret
       property: anotherKey # name of the field in the GCP Secret JSON, unused for plain values
 ```
+### Note
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:73]:
```suggestion
>**NOTE:** The trusted and untrusted files are only applied to trusted or untrusted clusters respectively. While the workload file is applied to both trusted and untrusted clusters.
```

File: prow/cluster/resources/external-secrets/README.md [hunk 2]
```diff
@@ -70,6 +70,10 @@ spec:
...
+The trusted/untrusted files are only applied to the trusted and untrusted clusters respectively, while the workload file is applied to both clusters.
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:74]:
```suggestion
   The presubmit and pj-tester jobs are executed on untrusted clusters, while the periodic jobs are run on trusted clusters. Adding a Secret to the proper file allows the user to specify which type of clusters should have access to the Secret.
```

File: prow/cluster/resources/external-secrets/README.md [hunk 3]
```diff
@@ -70,6 +70,10 @@ spec:
...
+
+The pre-submit and pj-tester jobs are executed on untrusted clusters, while the periodic jobs are run on the trusted clusters. Adding the secret to the proper file allows the user to specify when type of clusters should have access to the secret.
```
Comment (mmitoraj) [inline prow/cluster/resources/external-secrets/README.md:76]:
```suggestion

```

**Review by mmitoraj — COMMENTED** [review]
I added a few suggestions :)

**Review by mmitoraj — APPROVED** [review]

**Pull request merged**